### ISSUE TITLE
Plan ModelInfo replacement

### ISSUE THREAD

**Issue by baseTwo** [FirelyTeam/firely-cql-sdk#582]
CQL additions for operators, define types in cql, base type library for cql in cql.

**Comment by ewoutkramer** [comment]
There are two parts to it:
* The public discussion on what this should look like in CQL
* What our internal representation looks like.

When we will start #2 before we have #1, we might need to re-align when the public discussion is done. So we'll make do with our current xml-based modelinfo classes.

**Comment by baseTwo** [comment]
Triaged against `develop` at `3810aa572`. **Keeping — nothing here has been done, but it needs scoping before anyone can act on it.**

### Status of the subject matter: unchanged since 2024

`@ewoutkramer`'s comment split this into (1) the public CQL-community discussion about what a CQL-native type/model declaration should look like, and (2) the SDK's internal representation, with the conclusion "we'll make do with our current xml-based modelinfo classes" until (1) lands. That is still exactly where things stand:

- The XML-based ModelInfo is still the only representation: `Cql/Cql.Model/modelinfo.xsd`, the generated `Cql/Cql.Model/modelinfo.cs`, deserialised by `XmlSerializer` at `Cql/Cql.Model/Models.cs:13` / `:29-32`, from two embedded resources — `Fhir401` and `ElmR1` (`Cql/Cql.Model/Models.cs:15-19`).
- CQL 1.5.3 itself still has no way to declare types or a base type library in CQL (nothing in `/spec/cql/condensed/`), so part (1) remains external, upstream work that this repo cannot unblock on its own.

### What this issue is *not*

Per this issue's own framing plus #1531's cross-reference: **#582 is not a duplicate of #1531 and does not cover its content.** Confirmed by reading both. #1531 ("Retrieve's primary-code-path lookup is hard-coded to FHIR 4.0.1; no resolver reports its CQL model") is about plumbing a model reference through `TypeResolver` and removing the hard-coded `ModelMapping = Models.ClassesById(Models.Fhir401)` at `Cql/Cql.Compiler/CodeBuilderContext.StaticHelpers.cs:22-23`. #1531's own Summary already states "#582 is a one-line stub about CQL-in-CQL type libraries", and that assessment holds: this issue is about *replacing* ModelInfo with CQL-native declarations, not about correcting how the existing ModelInfo is consulted. No overlap to merge, no duplication to record.

Also distinct from #863 (add FHIR 5 modelinfo) and #407 (profile-informed ModelInfo), both of which stay inside the XML ModelInfo model rather than replacing it.

### Why it is only medium confidence

The body is one line — "CQL additions for operators, define types in cql, base type library for cql in cql" — with no acceptance criteria, no code pointers, and no definition of done. It is a placeholder by design (the maintainer comment says as much), not a stale ticket: it names no deleted symbols and has no dead links, so there is nothing to rewrite. But as written nobody can pick it up.

The honest options are (a) keep it as an explicitly-parked epic and re-scope it when the CQL community discussion produces something concrete, or (b) close it as "tracked upstream" and re-file when that happens. Per triage policy — never close on age or on a guess, default to keep — this stays open, but it should be labelled as parked/blocked-upstream and given at minimum: a link to the relevant HL7/CQL discussion thread, and a one-paragraph statement of what "ModelInfo replacement" means for this SDK's internals. Whoever owns the CQL spec relationship should make that call; a triage pass should not.

Confidence: medium — the subject matter is verifiably untouched and correctly distinct from #1531, #863 and #407, but the issue's own scope is too thin for a triage pass to judge whether it should live on as an epic or be tracked upstream instead.
